internal/pkg/cli/svc_deploy.go

```diff
@@ -591,7 +593,66 @@ func validateAlias(svcName, alias string, app *config.Application, envName strin
 `, color.HighlightCode("http.alias"), envName, app.Name, app.Domain, envName,
 		app.Name, app.Domain, app.Name, app.Domain, app.Name,
 		app.Domain, app.Domain, app.Domain)
-	return fmt.Errorf("alias is not supported in hosted zones not managed by Copilot")
+	return fmt.Errorf("alias is not supported in hosted zones that are not managed by Copilot")
+}
+
+func checkUnsupportedAlias(alias, envName string, app *config.Application) error {
+	var regEnvHostedZone, regAppHostedZone *regexp.Regexp
+	var err error
+	// Example: subdomain.env.app.domain, env.app.domain
+	if regEnvHostedZone, err = regexp.Compile(fmt.Sprintf(`^([^\.]+\.)?%s.%s.%s`, envName, app.Name, app.Domain)); err != nil {
+		return err
+	}
+
+	// Example: subdomain.app.domain, app.domain
+	if regAppHostedZone, err = regexp.Compile(fmt.Sprintf(`^([^\.]+\.)?%s.%s`, app.Name, app.Domain)); err != nil {
+		return err
+	}
+
+	if regEnvHostedZone.MatchString(alias) {
+		return errors.New("environment-level alias is not supported yet")
+	}
+
+	if regAppHostedZone.MatchString(alias) {
+		return errors.New("application-level alias is not supported yet")
+	}
+
+	if alias == app.Domain {
+		return errors.New("root domain alias is not supported yet")
+	}
+
+	return nil
+}
+
+func validateRDSvcAliasAndAppVersion(svcName, alias, envName string, app *config.Application, appVersionGetter versionGetter) error {
+	if alias == "" {
+		return nil
+	}
+	if err := validateAppVersion(app.Name, appVersionGetter); err != nil {
+		logAppVersionOutdatedError(svcName)
+		return err
+	}
+	// Alias should be within root hosted zone.
+	aliasInvalidLog := fmt.Sprintf(`%s should match the pattern <subdomain>.%s 
+Where <subdomain> cannot be the application name.
+`, color.HighlightCode("http.alias"), app.Domain)
```

Comment:
nit: (here and elsewhere)-- would it be more helpful to surface the invalid value rather than "http.alias"?